### PR TITLE
feat: add pusher beams provider

### LINKED ISSUE #4634: [NV-3052] 🚀 Feature: Add Pusher Beam as a Push Provider
### 🔖 Feature description

Use the [pusher beam docs](https://pusher.com/docs/beams/reference/server-sdk-node/) to create a new push provider while following [Novu's guide to a new provider](https://docs.novu.co/community/add-a-new-provider#how-to-add-a-new-provider)

### 🎤 Why is this feature needed ?

more options for the community

### ✌️ How do you aim to achieve this?

community contribution

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-3052](https://linear.app/novu/issue/NV-3052/🚀-feature-add-pusher-beam-as-a-push-provider)</sub>

### PR DESCRIPTION
**Pull request by im-perativa**
### What change does this PR introduce?

Add Pusher Beam as a Push Provider

### Why was this change needed?

Closes #4634

### CODE REVIEW

**Review by djabarovgeorge — COMMENTED** [review]
Looks good, i left a couple of questions regarding things i was not sure about.
Did you had a chance to test triggering notifications locally?

**Comment by im-perativa** [comment]
> Looks good, i left a couple of questions regarding things i was not sure about. Did you had a chance to test triggering notifications locally?

Hi @djabarovgeorge , sorry I didn't mean to get this reviewed before since I still mark this as draft PR but I think it's good for review now. Yes I have tested the trigger function locally.

File: libs/shared/src/consts/providers/channels/push.ts [hunk 1]
```diff
@@ -48,4 +55,12 @@ export const pushProviders: IProviderConfig[] = [
     logoFileName: { light: 'push-webhook.svg', dark: 'push-webhook.svg' },
     betaVersion: true,
   },
+  {
+    id: PushProviderIdEnum.PusherBeams,
+    displayName: 'Pusher Beams',
+    channel: ChannelTypeEnum.PUSH,
+    credentials: pusherBeamsConfig,
+    docReference: 'https://docs.novu.co/channels-and-providers/push/pusher-beams',
```
[the remark this answers is not in the document]
Comment (im-perativa) [reply libs/shared/src/consts/providers/channels/push.ts:63]: Sure, i opened the [issue](https://github.com/novuhq/docs/issues/292)

**Comment by Cliftonz** [comment]
@im-perativa look like you need to fix some conflicts

**Pull request closed without merging**